src/docfx/lib/path/BasePath.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Newtonsoft.Json;
+
+namespace Microsoft.Docs.Build
+{
+    [JsonConverter(typeof(BasePathJsonConverter))]
+    internal class BasePath
+    {
+        public string Original { get; set; }
+
+        private string RelativePath
+            => Original.StartsWith('/') ? Original.TrimStart('/') : Original;
```

Comment:
return "." for empty path so callers can pass this to `Path` functions, set the value in constructor for better performance

Reply:
consider override `ToString() => Original`

Reply:
> return "." for empty path so callers can pass this to `Path` functions, set the value in constructor for better performance

`Path.Combine(".", "test")` would output `.\test`